src/backend/distributed/metadata/dependency.c

```diff
@@ -1101,10 +1089,45 @@ BuildViewDependencyGraph(Oid relationId, HTAB *nodeMap)
 		}
 	}
 
+	return node;
+}
+
+
+/*
+ * GetPgDependTuplesForDependingObjects scans pg_depend for given object and
+ * returns a list of heap tuples for the objects depending on it.
+ */
+List *
+GetPgDependTuplesForDependingObjects(Oid targetObjectClassId, Oid targetObjectId)
```

Comment:
Can you choose a different class for an object?

Reply:
I couldn't find a way to determine class of an object given the oid, do you know a way to know this ?